tools/witx/src/ast.rs

```diff
@@ -209,12 +211,27 @@ impl Type {
 
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 pub enum BuiltinType {
-    String,
-    Char8,
-    USize,
-    U8,
+    Char,
+    U8 {
+        /// Indicates whether this type is intended to represent the `char`
+        /// type in the C language. The C `char` type is often unsigned, but
+        /// it's language-specific. At an interface-types level this is an
+        /// unsigned byte but binding generators may wish to bind this as the
+        /// language-specific representation for a C character instead.
+        lang_c_char: bool,
```

Comment:
The `char8` type in witx is meant to roughly correspond to [`char8_t`](https://en.cppreference.com/w/cpp/language/types#Character_types). Besides signedness, `char8` indicates an intent to point to Unicode string data.

Reply:
I added a comment along these lines, but did you think we needed other changes? I'm hoping that we can eliminate the `char8` type as we move closer to pure-interface-types wasi.

Reply:
Agreed; we should ideally use interface-types' `string` type for strings wherever possible.